project/src/com/google/daggerquery/executor/models/Graph.java

```diff
@@ -0,0 +1,44 @@
+/*
+Copyright 2020 Google LLC
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    https://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+ */
+
+package com.google.daggerquery.executor.models;
+
+import java.util.List;
+import java.util.Set;
+
+/**
+ * The interface representing the graph created by the Dagger SPI plugin.
+ *
+ * <p>Provides the methods needed to traverse this graph and to execute queries.
+ */
+public interface Graph {
+
+  /**
+   * Returns all node's dependencies.
+   */
+  List<String> getDependencies(String node);
+
+  /**
+   * Checks if a node is presented in the graph or not.
+   */
+  Boolean containsNode(String node);
+
+  /**
+   * Returns all the nodes presented in the graph.
+   */
+  Set<String> getAllNodes();
```

Comment:
ditto, would it be possible to return `ImmutableSet`?

Reply:
👌

Reply:
![pokemon](https://user-images.githubusercontent.com/19249980/91904520-bb0e5f00-ecad-11ea-8fe0-05562b55ed91.jpeg)

Reply:
Done 😎